mitodl/ocw-studio
# Added support for soft/hard deletes of content

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of #209 

#### What's this PR do?
This enables deletion of `WebsiteContent` and integrates updates to GitHub by performing deletes as soft deletes, following by an asynchronous hard delete after the deletion has been synced to GitHub.

#### How should this be manually tested?
Find a piece of content for a website and go to the browable API for that (e.g. `http://ocw-studio.odl.local:8043/api/websites/9-012-the-brain-and-cognitive-sciences-ii-spring-2002/content/3ba4040cbae19e58b113c7fd89033f53/`). Click the delete option, and confirm that soft deletes the record and then triggers a task that makes a commit to delete the content.
